### ISSUE TITLE
teardown() is called twice

### ISSUE THREAD

**Issue by forrest79** [nette/tester#266]
In version 1.6 we have this problem. Our DB abstraction has some of the following code (simplified):

```php
$resource = @pg_query_params($this->getResource(), $sql, $variables);

if (is_resource($resource) === FALSE) {
	throw new Database\QueryException('Database query failed. ' . $this->getLastError());
}
```

For our DB tests, we begin transaction in setUp() and we rollback it in tearDown(). If we want to test some QueryException via Assert::exception(...), in Tester 1.6 is called tearDown before generating our QueryExceptionm because @pg_query_params generate WARNING or NOTICE (which is suppressed by @) and this called new $errorHandler from Framework/TestCase.php line 109. This close transaction in our code and I get bad error message (because I don't have last error...) and after that is called tearDown properly from line 143.

**Comment by milo** [comment]
@forrest79 If you can, could you try #267?

**Comment by jiripudil** [comment]
@milo :+1: could this fix make it into a tagged stable release anytime soon?

**Comment by milo** [comment]
@jiripudil @forrest79 Released 1.6.1

**Comment by forrest79** [comment]
Works good, thanks :-)
